mediapipe/examples/desktop/face_mesh_dll/face_mesh_cpu.cpp

```diff
@@ -0,0 +1,69 @@
+#include "face_mesh_lib.h"
+
+int main(int argc, char **argv) {
+  google::InitGoogleLogging(argv[0]);
+  absl::ParseCommandLine(argc, argv);
+
+  cv::VideoCapture capture;
+  capture.open(0);
+  if (!capture.isOpened()) {
+    return -1;
+  }
+
+  constexpr char kWindowName[] = "MediaPipe";
+
+  cv::namedWindow(kWindowName, /*flags=WINDOW_AUTOSIZE*/ 1);
+#if (CV_MAJOR_VERSION >= 3) && (CV_MINOR_VERSION >= 2)
+  capture.set(cv::CAP_PROP_FRAME_WIDTH, 640);
+  capture.set(cv::CAP_PROP_FRAME_HEIGHT, 480);
+  capture.set(cv::CAP_PROP_FPS, 30);
+#endif
+
+  LOG(INFO) << "VideoCapture initialized.";
+
+  MPFaceMeshDetector *faceMeshDetector = FaceMeshDetector_Construct();
+
+  LOG(INFO) << "FaceMeshDetector constructed.";
+
+  LOG(INFO) << "Start grabbing and processing frames.";
+  bool grab_frames = true;
+
+  while (grab_frames) {
+    // Capture opencv camera.
+    cv::Mat camera_frame_raw;
+    capture >> camera_frame_raw;
+    if (camera_frame_raw.empty()) {
+      LOG(INFO) << "Ignore empty frames from camera.";
+      continue;
+    }
+    cv::Mat camera_frame;
+    cv::cvtColor(camera_frame_raw, camera_frame, cv::COLOR_BGR2RGB);
+    cv::flip(camera_frame, camera_frame, /*flipcode=HORIZONTAL*/ 1);
```

Comment:
so we are getting upside-down flipped images from the camera?

Reply:
No, they flip an image horizontally (`/*flipcode=HORIZONTAL*/ 1`), and that is done because mediapipe will detect keypoint from `left top corner` of the image when OpenCV is using `right top corner` as a center of the coordinate system.

Reply:
This part I take from example: `mediapipe\examples\desktop\demo_run_graph_main.cc`.

Reply:
hm, strange. OpenCV c.s. starts at top left corner for sure. Perhaps they are undoing mirroring effect of the camera, but then again I don't see a reason to do this.